lib/ddtrace/configuration/base.rb

```diff
@@ -37,7 +37,7 @@ def settings(name, &block)
 
         def new_settings_class(&block)
           Class.new { include Datadog::Configuration::Base }.tap do |klass|
-            klass.instance_eval(&block) if block_given?
+            klass.instance_eval(&block) if block
```

Comment:
This was changed due to performance, TIL: https://docs.rubocop.org/rubocop-performance/cops_performance.html#performanceblockgivenwithexplicitblock

Reply:
Is `block_given?` less performant than checking the `block` variable?

Reply:
Yeah, it was news to me. When the block parameter is defined, `def foo(&block)`, `if block` is measurably faster then `block_given?`. When `&block` is not defined, using `block_given?` is pretty fast already:

```ruby
require 'benchmark/ips'

def block(&block)
  0 if block
end

def given(&block)
  0 if block_given?
end

def implicit
  0 if block_given?
end

Benchmark.ips do |x|
  x.config(time: 2, warmup: 0.1) if x.respond_to? :config

  x.report('[explicit] if block', 'block')
  x.report('[explicit] if block_given?', 'given')
  x.report('[implicit] if block_given?', 'implicit')

  x.compare!
end
```

```
Comparison:
[implicit] if block_given?: 22882424.6 i/s
[explicit] if block:        21200506.5 i/s - 1.08x  (± 0.00) slower
[explicit] if block_given?: 15820249.6 i/s - 1.45x  (± 0.00) slower
```